overlord/snapstate/autorefresh.go

```diff
@@ -278,6 +278,16 @@ func (m *autoRefresh) Ensure() error {
 		return err
 	}
 
+	// is there a previously partially inhibited auto-refresh that can now be continued?
+	if canContinueAutoRefresh(m.state) {
+		err := m.launchAutoRefresh()
+		if err == nil {
+			// clear the continue flag if the auto-refresh was scheduled successfully
+			m.state.Cache("continue-auto-refresh", nil)
```

Comment:
if we error and do not clear this we might end up trying again very soon, we might have toconsider the logic for 

```
// Check that we have reasonable delays between attempts.
// If the store is under stress we need to make sure we do not
// hammer it too often
```

even for this launchAutoRefresh

maybe some bit of that needs to be split/shared somehow to avoid repeating it twice

Reply:
The retry delay is 20min so we should probably make an exception for the first auto-refresh after a pre-download. Otherwise, we notify the user and after the snap closes, there's still a long time before the auto-refresh is attempted which is confusing. Maybe the "continue-auto-refresh" can be a counter instead (like a retry id) and when it's zero we ignore the retryDelay. wdyt?

Reply:
@pedronis

Reply:
making it a counter seems reasonable, not sure about starting at 0 though, it could start at 1, it's a also probably ok if the monitor code on success just sets to 1 even if it was higher. The issue is mostly about immedate retries when we had an error